src/langcheck/metrics/en/reference_based_text_quality.py

```diff
@@ -70,96 +70,18 @@ def semantic_similarity(
         'local', 'openai', 'azure_openai'
     ], ('Unsupported embedding model type. '
         'The supported ones are ["local", "openai", "azure_openai"]')
-    batch_size = 8
     if model_type == 'local':
         # The 'all-mpnet-base-v2' model has the highest average performance out
         # of all the existing sentence-transformer models that have been
         # evaluated.
         # Ref: https://www.sbert.net/docs/pretrained_models.html#model-overview
-        model = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')
-        generated_embeddings = []
-        reference_embeddings = []
-        for i in tqdm_wrapper(range(0, len(generated_outputs), batch_size),
-                              total=(len(generated_outputs) + batch_size - 1) //
-                              batch_size,
-                              desc='Getting embeddings'):
-            batch_generated_outputs = generated_outputs[i:i + batch_size]
-            batch_reference_outputs = reference_outputs[i:i + batch_size]
-            batch_generated_embeddings = model.encode(batch_generated_outputs)
-            batch_reference_embeddings = model.encode(batch_reference_outputs)
-            generated_embeddings.extend(batch_generated_embeddings)
-            reference_embeddings.extend(batch_reference_embeddings)
+        scorer = SentenceTransformerSimilarityScorer(language='en')
     else:  # openai or azure_openai
-        # Initialize the openai object if openai_client is None
-        # TODO: Refactor this into OpenAIBasedEvaluator?
-        if not openai_client:
-            if model_type == 'openai':
-                openai_client = OpenAI()
-            else:  # azure_openai
-                # https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/migration?tabs=python-new%2Cdalle-fix#embeddings
-                openai_client = AzureOpenAI(
-                    api_key=os.getenv("AZURE_OPENAI_KEY"),
-                    api_version=os.getenv("OPENAI_API_VERSION"),
-                    azure_endpoint=os.getenv(
-                        "AZURE_OPENAI_ENDPOINT"))  # type: ignore
-        if model_type == 'azure_openai' and not openai_args:
-            raise AssertionError(
-                'The embedding model deployment must be specified in '
-                '`openai_args` for the azure_openai type, e.g. '
-                '`openai_args={"model": "YOUR_DEPLOYMENT_NAME"}`')
```

Comment:
I think we should keep this check?

Reply:
Removed it totally by accident. Thanks for the good catch.